api/response/schema.py

```diff
@@ -15,40 +14,48 @@ class Meta:
         model = ResponseModel
 
 
+class ResponseInputs(graphene.InputObjectType):
+    question_id = graphene.Int(required=True)
+    rate = graphene.Int()
+    text_area = graphene.String()
+    missing_items = graphene.List(graphene.Int)
+
+
 class CreateResponse(graphene.Mutation):
     class Arguments:
-        question_id = graphene.List(graphene.Int, required=True)
+        responses = graphene.List(
+            ResponseInputs, required=True
+        )
         room_id = graphene.Int(required=True)
-        rate = graphene.Int()
-        text_area = graphene.String()
-        missing_items = graphene.List(graphene.Int)
 
     response = graphene.List(Response)
 
     def mutate(self, info, **kwargs):
         validate_empty_fields(**kwargs)
         query = Room.get_query(info)
+        responses = []
+        errors = []
         room = query.filter_by(id=kwargs['room_id']).first()
         if not room:
             raise GraphQLError("Non-existent room id")
-        question_ids = kwargs.pop('question_id')
-        responses = []
-        questions = Question.get_query(info).filter(
-            QuestionModel.id.in_(question_ids)).all()
-        valid_question_ids = set()
-        for question in questions:
-            valid_question_ids.add(question.id)
+        for each_response in kwargs['responses']:
+            question = QuestionModel.query.filter_by(
+                id=each_response.question_id).first()
+            if not question:
+                errors.append(
+                    "Response to question {} was not saved because it does not exist".format( # noqa
```

Comment:
since we end up having # noqa either way I think we should just have one line since the # noqa being inside the parentheses doesn't give a good picture.

Reply:
Okay, good idea.